### PR TITLE
feat: remove all TypeScript references when `isTypeScript` is passed as `false` to `remix.init`

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Just like https://github.com/remix-run/indie-stack/pull/127 & https://github.com/remix-run/blues-stack/pull/57

### CODE REVIEW

**Comment by machour** [comment]
#47 was merged 👍🏼

**Comment by williamrjribeiro** [comment]
Hello! So much time has passed and this stack is still broken when choosing JavaScript instead of TypeScript 😿. I know this PR is still a `Draft` but I'll start adding feedback already. I just wanna use JS with Remix 💿

File: remix.init/index.js [hunk 1]
```diff
@@ -2,26 +2,80 @@ const { execSync } = require("child_process");
...
+    (need) => need !== "typecheck"
+  );
 
-function getRandomString(length) {
-  return crypto.randomBytes(length).toString("hex");
-}
+  return [fs.writeFile(deployWorkflowPath, YAML.stringify(deployWorkflow))];
+};
+
+const cleanupVitestConfig = (vitestConfig, vitestConfigPath) => {
```
Comment (williamrjribeiro) [inline remix.init/index.js:20]: If `vite-tsconfig-paths` is not needed, it should be removed.

File: remix.init/index.js [hunk 2]
```diff
@@ -2,26 +2,80 @@ const { execSync } = require("child_process");
...
+const readFileIfNotTypeScript = (
+  isTypeScript,
+  filePath,
+  parseFunction = (result) => result
+) =>
+  (isTypeScript ? Promise.resolve() : fs.readFile(filePath, "utf-8")).then(
+    parseFunction
+  );
+
+const removeUnusedDependencies = (dependencies, unusedDependencies) =>
```
Comment (williamrjribeiro) [inline remix.init/index.js:40]: This is a cool idea that could be used for the inverse if needed: adding new JS dependencies. 💡

File: remix.init/index.js [hunk 3]
```diff
@@ -30,59 +84,57 @@ async function main({ rootDirectory, isTypeScript }) {
...
+  ];
+
+  if (!isTypeScript) {
+    fileOperationPromises.push(
+      ...cleanupDeployWorkflow(deployWorkflow, DEPLOY_WORKFLOW_PATH)
+    );
+
+    fileOperationPromises.push(
+      ...cleanupVitestConfig(vitestConfig, VITEST_CONFIG_PATH)
+    );
```
Comment (williamrjribeiro) [inline remix.init/index.js:134]: The Cypress `ESLint` configurations are not working. I tried using [`eslint-plugin-cypress`](https://github.com/cypress-io/eslint-plugin-cypress#readme) and it just worked! Actually, it detected an unwanted arbitrary `wait()`. 😓

File: remix.init/index.js [hunk 4]
```diff
@@ -30,59 +84,57 @@ async function main({ rootDirectory, isTypeScript }) {
...
-    fs.rm(path.join(rootDirectory, ".github/PULL_REQUEST_TEMPLATE.md")),
-  ]);
+    fs.rm(path.join(rootDirectory, ".github", "PULL_REQUEST_TEMPLATE.md")),
+  ];
+
+  if (!isTypeScript) {
+    fileOperationPromises.push(
+      ...cleanupDeployWorkflow(deployWorkflow, DEPLOY_WORKFLOW_PATH)
+    );
+
```
Comment (williamrjribeiro) [inline remix.init/index.js:131]: In `remix.config.js`, change to `server: "./server.js",` and 
`path.relative(appDir, "cypress/support/test-routes/create-user.ts")`

File: remix.init/index.js [hunk 5]
```diff
@@ -30,59 +84,57 @@ async function main({ rootDirectory, isTypeScript }) {
...
+    fileOperationPromises.push(
+      ...cleanupVitestConfig(vitestConfig, VITEST_CONFIG_PATH)
+    );
+  }
+
```
Comment (williamrjribeiro) [inline remix.init/index.js:136]: Finally, in `cypress.config.js` changing to `module.exports` format started working again.

**Review by williamrjribeiro — COMMENTED** [review]
I hope my comments help this PR to be concluded and merged sooner. 🙏 

Cheers!

File: remix.init/index.js (hunk 1 above)
Comment (MichaelDeBoey) [reply remix.init/index.js:20]: It is needed

File: remix.init/index.js (hunk 2 above)
Comment (MichaelDeBoey) [reply remix.init/index.js:40]: We prefer to remove the dependencies afterwards, as that makes it easier to update versions

File: remix.init/index.js (hunk 3 above)
Comment (MichaelDeBoey) [reply remix.init/index.js:133]: `eslint-plugin-cypress` is added in this PR

File: remix.init/index.js (hunk 5 above)
Comment (MichaelDeBoey) [reply remix.init/index.js:135]: This will be fixed once https://github.com/remix-run/remix/pull/4030 is released

**Comment by MichaelDeBoey** [comment]
@williamrjribeiro @mcansh This one is good to go I think 👍

**Comment by mcansh** [comment]
> @williamrjribeiro @mcansh This one is good to go I think 👍

can you fix the merge conflict? 😬

**Comment by MichaelDeBoey** [comment]
@mcansh You were too fast 🙈

**Comment by mcansh** [comment]
> @mcansh You were too fast 🙈

just doing one final check before logging off 😆

**Comment by MichaelDeBoey** [comment]
@mcansh I added `eslint-plugin-cypress` to the wrong `package.json` 😅
Fixed now too, so ESLint will be OK now 👍

**Review by mcansh — APPROVED** [review]

**Pull request merged**